snapcraft/plugins/catkin.py

```diff
@@ -127,77 +160,59 @@ def gcc_version(self):
     @property
     def rosdir(self):
         return os.path.join(self.installdir,
-                            'opt', 'ros', self.options.rosversion)
+                            'opt', 'ros', self.options.rosdistro)
 
-    def _deps_from_packagesxml(self, f, pkg):
-        tree = lxml.etree.parse(f)
+    def _find_dependencies(self):
+        if self.package_deps_found:
+            return
```

Comment:
It would be good to explain this in a comment, because it's weird. What are the cases that makes us call find_dependencies twice?